### ISSUE TITLE
Test console output not matching the observed behaviour

### ISSUE THREAD

**Issue by dhood** [ros2/build_farmer#114]
I noticed a peculiar talker-listener test failure (flaky).

The listener receives its message, but the talker doesn't output to console that it published anything, so the output matching logic fails.

https://ci.ros2.org/view/nightly/job/nightly_win_rep/1057/testReport/(root)/projectroot/test_tutorial_talker_listener__rmw_fastrtps_cpp/
```
(test_executable_0) pid 10252: ['C:/J/workspace/nightly_win_rep/ws/build/demo_nodes_cpp/Release/talker.exe'] (all > console, InMemoryHandler: test_executable_0)
(test_executable_1) pid 7536: ['C:/J/workspace/nightly_win_rep/ws/build/demo_nodes_cpp/Release/listener.exe'] (all > console, InMemoryHandler: test_executable_1)
[test_executable_1] I heard: [Hello World: 1]
(test_executable_1) rc 1
() tear down
(test_executable_0) signal SIGTERM
(test_executable_0) rc 1
```

The usual output is [something like](https://ci.ros2.org/view/nightly/job/nightly_win_rep/850/testReport/(root)/test_talker_listener__rmw_fastrtps_cpp_Release/test_executable/):
```
(test_executable_0) pid 7412: ['C:/J/workspace/nightly_win_rep/ws/build/demo_nodes_cpp/Release/talker.exe'] (all > console, InMemoryHandler: test_executable_0)
(test_executable_1) pid 4648: ['C:/J/workspace/nightly_win_rep/ws/build/demo_nodes_cpp/Release/listener.exe'] (all > console, InMemoryHandler: test_executable_1)
[test_executable_0] Publishing: 'Hello World: 1'
[test_executable_0] Publishing: 'Hello World: 2'
[test_executable_1] I heard: [Hello World: 2]
(test_executable_1) rc 1
() tear down
(test_executable_0) signal SIGTERM
(test_executable_0) rc 1
```
@dirk-thomas I think this only started happening since april 21 after the colcon switch, and I know there have been other console output capturing peculiarities; any ideas come to mind?

**Comment by dirk-thomas** [comment]
The output handling of the different processes happens in `launch`. Neither `colcon` nor `pytest` can distinguish between them. So I don't have any idea why that would have changed.

**Comment by dhood** [comment]
There's a chance that the publisher really didn't publish its message and that the listener received it from somewhere else, but that at least in our "old" CI setup would not have been possible. If indeed it is from another node, it would have to be from a freshly started talker to receive message with data 1 (three days in a row). Would connext and fastrtps be running their tests in parallel or anything?

**Comment by dirk-thomas** [comment]
That could be related to and be addressed by ros2/ci#154. Until that patch it was running tests from different packages in parallel.

**Comment by dhood** [comment]
@nuclearsandwich did this pop up again after https://github.com/ros2/ci/pull/154?
